sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -380,29 +404,44 @@ private HttpPipeline setupPipeline(HttpClient httpClient) {
      */
     public Mono<MsalToken> authenticateWithUsernamePassword(TokenRequestContext request,
                                                             String username, String password) {
-        return Mono.fromFuture(publicClientApplication.acquireToken(
+        return Mono.fromFuture(() -> getPublicClientApplication(false).acquireToken(
             UserNamePasswordParameters.builder(new HashSet<>(request.getScopes()), username, password.toCharArray())
                 .build()))
-            .map(ar -> new MsalToken(ar, options));
+            .onErrorResume(t -> Mono.error(new ClientAuthenticationException("Failed to acquire token with username "
```

Comment:
Should this be `onErrorMap`? This logic looks to be more of an error mapping than an alternate resume from an exception given this will propagate an error.

Reply:
Yes - I'm not aware of this operator, changing them now